File: 프로그래머스/unrated/120894. 영어가 싫어요/영어가 싫어요.py
def solution(numbers):
    # 영어 단어와 해당하는 정수를 매핑하는 사전 생성
    word_to_number = {
        "zero": 0, "one": 1, "two": 2, "three": 3, "four": 4,
        "five": 5, "six": 6, "seven": 7, "eight": 8, "nine": 9
    }

    result = 0  # 결과를 저장할 변수 초기화
    current_word = ""  # 현재 읽고 있는 영어 단어 저장용 변수 초기화

    # 문자열을 순회하면서 영어 단어를 읽어서 정수로 변환
    for char in numbers:
        current_word += char  # 현재 단어에 글자 추가

        # 현재 단어가 영어 단어 사전에 있을 때
        if current_word in word_to_number:
            result = result * 10 + word_to_number[current_word]
            current_word = ""  # 현재 단어 초기화

    return result

File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.java
class Solution {
    public long solution(String numbers) {
        long answer = 0;
        
        // replace 함수로대체
        numbers = numbers.replace("zero","0");
        numbers = numbers.replace("one","1");
        numbers = numbers.replace("two","2");
        numbers = numbers.replace("three","3");
        numbers = numbers.replace("four","4");
        numbers = numbers.replace("five","5");
        numbers = numbers.replace("six","6");
        numbers = numbers.replace("seven","7");
        numbers = numbers.replace("eight","8");
        numbers = numbers.replace("nine","9");

        answer = Long.parseLong(numbers);
        // int type으로 변환 되엇으니 다시 long 으로 변환
        return answer;
    }
}